notify/notify.go

```diff
@@ -640,6 +640,17 @@ func (n *DedupStage) Exec(ctx context.Context, _ log.Logger, alerts ...*types.Al
 	}
 
 	if n.needsUpdate(entry, firingSet, resolvedSet, repeatInterval) {
+		if entry != nil {
```

Comment:
why not moving this verification to needsUpdate()?

Reply:
Initially I wanted to do that there. I had an idea to emit a log message only when needsUpdate is positive (to indicate race). To achieve that in needsUpdate I would have to rewrite the method, which I thought would bring less clarity. I can refactor that method, if you think that will be ok for understanding the change.